File: ruby/grains/grains.rb
class Grains
  TOTAL_NUMBER_OF_SQUARES = 64
  def self.square(square)
    raise ArgumentError if square <= 0 || square > 64
    2**(square - 1)
  end

  def self.total
    2**TOTAL_NUMBER_OF_SQUARES - 1
  end
end

File: javascript/grains/grains.js
const totalNumberOfSquares = 64

export const square = (number) => {
  if (number <= 0 || number > totalNumberOfSquares) throw new Error('square must be between 1 and 64')
  return BigInt(Math.pow(2, number - 1))
};

export const total = () => {
  return BigInt(Math.pow(2, 64)) - 1n
};
